Add release post for 10.11.0 (#1615)
jellyfin/jellyfin.org: blog/2025/10-19-jellyfin-release-10.11.0/index.mdx

---
title: Jellyfin 10.11.0
authors:
  - joshuaboniface
slug: jellyfin-release-10.11.0
tags: [release, server]
---

> We are pleased to bring you Jellyfin 10.11.0, our new stable release. This is probably one of, if not the, biggest and most impactful releases we've done yet, with some massive backend changes to improve performance and long-term expandability and maintainability. This release has been a long time coming, with over 6 months of development and another 6 months of RC testing, throwing our planned 6-month release schedule completely out of whack, but we definitely think the results are worth it - both for users right now, and for the long-term health of the project.

Reviewer: >to improve performance

didn't we say there will be a minor performance degradation for 10.11 compared to 10.10 in most cases?

0. As always for major upgrades, ensure you **STOP Jellyfin and take a FULL BACKUP OF YOUR DATA AND CONFIG DIRECTORIES before upgrading**! While the upgrade process _should_ be seamless and has guardrails, we've seen a lot of strange bugs during RC testing and it's better to be safe than sorry.

1. You **MUST** be running Jellyfin 10.10.7 before upgrading to 10.11.0! You **may** be fine with Jellyfin 10.9.11 but this is less-extensively tested. **Upgrading from any other versions is NOT supported and WILL fail**; upgrade to 10.10.7 **first**, then upgrade to 10.11.0.

2. The initial upgrade will include **MULTIPLE LONG-RUNNING MIGRATIONS** that may take up to several hours depending on your library size and state. **DO NOT CANCEL OR INTERRUPT THE SYSTEM** during these migrations; let them run overnight if possible on very large libraries. You can use the new Startup UI and Log Viewer (see below) from your local network to view the progress during the upgrade, and please see the section below for more details on "why".

> 3. The upgrade will make a backup of your existing `library.db` file that can be used to recover should the upgrade fail. Once you have successfully upgraded and Jellyfin 10.11.0 is running smoothly, you may delete this backup. If you need to try the migrations again due to a failure, stop Jellyfin, rename this file back to `library.db`, then start Jellyfin again, and the migration will be re-attempted.

Reviewer: should say what the backup file will be named

![Startup UI](startup-ui.png)

### Internal Backup & Restore Support

One of the first big benefits of the new backend is a very eagerly-awaited feature: internal, live Backup and Restore support! You can now take a full snapshot of your metadata database, back it up to external storage, and restore from it in the event of a major fault. You can also take regular backups just in case. This will greatly help both normal day-to-day maintenance, as well as upgrades in the future, always knowing that you have a good copy of the data.

The backup and restore system can only restore systems on which the backup was originally made, so this is not a tool to migrate from other OS's or 3rd party containers.

![Backups](backups.png)

### Aggressive in-memory Database Caching

> The new database engine is tuned to aggressive cache metadata in memory, to prevent slow disk reads from bottlenecking the system; this was a frequent problem in previous versions of Jellyfin and a point of regular complaint. In practice, this means that Jellyfin will use more RAM than before, up to the full size of your Library database. This RAM is not wasted however, as the speed-up is very noticeable, especially with large libraries, and the system will try to be as efficient as possible if other tasks require memory. Be aware of this as you upgrade, as it will look like a lot more resource utilization.

Reviewer: ```suggestion
The new database engine is tuned to aggressively cache metadata in memory, to prevent slow disk reads from bottlenecking the system; this was a frequent problem in previous versions of Jellyfin and a point of regular complaint. In practice, this means that Jellyfin will use more RAM than before, up to the full size of your Library database. This RAM is not wasted however, as the speed-up is very noticeable, especially with large libraries, and the system will try to be as efficient as possible if other tasks require memory. Be aware of this as you upgrade, as it will look like a lot more resource utilization.
```
we should also mention that the memory will be given back to the system if something else asks for the memory.

### Improved Locking Behaviour

The new database engine also enables several new locking modes; for most users, this is irrelevant, but users with frequent issues related to database locking can now tweak the settings. **More detail about this functionality can be found [in this blog post](/blog/SQLite-locking).**

## Removal of ARM32 (armhf) support

> In Jellyfin 10.10.0, we deprecated ARM32 support and have fully removed it from Jellyfin 10.11.0; Jellyfin now requires an ARM64 operating system to run on ARM platforms. In practical terms, this means dropping support for very old or very low-end ARM SBCs (Raspberry Pi 1/2, etc.). We found consistently that these very old SBCs were absolutely terrible at running Jellyfin, even without transcoding, and we decided that dropping support would streamline our release processes and help ensure that users get the best experience out of the box. If you are still running on one of these systems, or on a newer SBC that isn't running a 64-bit ARM operating system, you will need to upgrade your hardware and/or reinstall an arm64 operating system in order to upgrade to Jellyfin 10.11.0.

Reviewer: ```suggestion
In Jellyfin 10.10.0, we deprecated ARM32 support and have fully removed it from Jellyfin 10.11.0; Jellyfin now requires a 64-bit operating system for all architectures. In practical terms, this means dropping support for very old or very low-end ARM SBCs (Raspberry Pi 1/2, etc.). We found consistently that these very old SBCs were absolutely terrible at running Jellyfin, even without transcoding, and we decided that dropping support would streamline our release processes and help ensure that users get the best experience out of the box. If you are still running on one of these systems, or on a newer SBC that isn't running a 64-bit operating system, you will need to upgrade your hardware and/or reinstall a 64-bit operating system in order to upgrade to Jellyfin 10.11.0.
```

## Free Space Checks

Jellyfin now actively checks the available free space for its configuration and data directories. If you have less than 2GB of free space in each, Jellyfin now refuses to start to prevent data corruption. Additionally, checks were implemented to prevent certain path misconfigurations that are know to cause issues.

## New Features & Enhancements

We have several new features and enhancements to look forward to:

### User Experience

* Our search performance has been greatly improved, making it much quicker to find what you need.
* The Favourites page will now list Live TV channels, Music Videos, Photo Albums, Photos, and Seasons in addition to the existing media types.

> * We now have HEVC decoding support for Firefox 134+.

Reviewer: Should link to the client support table. There is a PR targeting that.

* Native styling of subtitles can be disabled.
* Shows can be displayed grouped in collections when viewing libraries.
* Theme media will now play in random order.
* Guest stars can be assigned roles, e.g., Kelsey Grammer as Captain Morgan Bateson.
* A customizable "splash screen" image can be set as a background on the login page.
* The experimental layout has a redesigned navigation and controls for filtering and sorting libraries.

### Administrator Experience

* The dashboard now displays stats for the number of media items on the server and available storage space.

> * Media segment providers can be configured per library similar to other metadata providers.

Reviewer: ```suggestion
* Media segment providers can now be configured per library similar to other metadata providers.
```

> * AV1 support can be enabled for VideoToolbox hardware acceleration.

Reviewer: ```suggestion
* AV1 decoding can be enabled for VideoToolbox hardware acceleration.
```

* M3U tuners can be configured to use the native framerate.
* Branding options are greatly expanded, including "splash screen" image support, extended disclaimer support, and custom CSS; even more is planned for the future to allow administrators to fully customize their user experience to your branding.
* The migration of dashboard pages to our new tech stack is nearly complete with most pages being converted, offering a much simpler and clearer organizational layout.
* When replacing files, Jellyfin will now recognize this and prune related data (chapters, trickplay files and extracted subtitles and attachments) on scan, making sure that these data will be re-generated, matching the new file.

### Transcoding and Media Handling

* We have moved to the new upstream FFmpeg 7.1, which brings many improvements.
* We now support 3D LUT-based HDR tone-mapping on Intel HD/UHD graphics and Rockchip RK3588, improving performance.

> * We now support Dolby Vision Profile 5 tone-mapping on Rockchip RK3588.

Reviewer: ```suggestion
* We now support Dolby Vision Profile 5 tone-mapping on Rockchip RKMPP hardware acceleration.
```
